doc/kdump/kdump_Remote_SSH_HLD.md

```diff
@@ -0,0 +1,228 @@
+    - [Functional Requirements  <a name="functional-requirements"></a>](#functional-requirements-a-namefunctional-requirementsa)
+    - [Configuration and Management Requirements](#configuration-and-management-requirements)
+	- [SSH Key Generation Requirements](#ssh-key-generation-requirement)
+- [kdump Remote Architechture](#kdump-remote-architechture)
+- [Functional Description](#functional-description)
+    - [Design Changes](#design-changes)
+- [Configurations and Management](#configuration-and-management)
+    - [CONFIG_DB Enhancements](#config_db-enhancements)
+    - [SAI API](#sai-api)
+    - [CLI/Yang MOdel Enhancements](#cliyang-model-enhancements)
+- [CLI Enhancements](#cli-enhancements)
+    - [CLI Configuration Commands](#cli-configuration-commands)
+    - [CLI Show Commands](#cli-show-commands)
+- [YANG Enhancement](#yang-enhancements)
+- [Warmboot Requirements](#warmboot-requirements)
+- [Test](#test)
+    - [Unit Test cases](#unit-test-cases)
+- [Links](#links)
+
+<!-- /TOC -->
+
+## List of Tables
+
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+## Revision
+
+Rev   |   Date   |  Author   | Change Description
+:---: | :-----:  | :------:  | :---------
+0.1   | 06/05/2024 | Ghulam Bahoo, Muhammad Ali Hussnain | Initial version
+## Overview
+This document outlines the configuration and usage of the kdump remote feature with ssh for the SONiC.
+
+## Scope
+
+This document describes how to configure remote kdump feature in SONiC infrastructure.
+
+## Definitions/Abbreviations
+
+### Table 1: Abbreviations
+
+| **Term**     | **Meaning**            |
+|  ----------- | ---------------------- |
+| SSH          | Secure Shell           |
+| kdump        | Kernel Dump            |
+| NFS          | Network File System    |
+
+## Introduction
+Kdump, a built-in Linux kernel feature, generates and stores a crash dump file in the event of a kernel panic. Currently SONiC lacks the functionality of storing dump files on a remote server. It offloads the storage from switch by saving the crash reports to a separate designated server for offline analysis. This feature extends existing kdump feature by enabling remote dumps via ssh protocol, allowing you to transfer kernel crash data to a designated remote server.
+
+## Requirements Overview  <a name="requirements-overview"></a>
+
+### Functional Requirements  <a name="functional-requirements"></a>
+This section describes the SONiC requirements for kdump remote feature.
+
+At a high level the following should be supported:
+1. The kernel core dump files must be stored on the a remote ssh server.
+### Configuration and Management Requirements
+
+- CLI support for configuring remote kdump feature enable/disable via ssh.
+- CLI support for configuring username and hostname of ssh server (username@server_address).
+- CLI support for configuring SSH private key path for ssh server (SSH_private_Key_Path).
+- CLI support for displaying crededentials of ssh server.
```

Comment:
typo. crededentials, please do a spell-checker for this HLD.

Reply:
Updated